apps/copilot-api/routes/auth.ts

```diff
@@ -0,0 +1,112 @@
+import express from 'express'
+import type {Request, Response} from 'express'
+import dotenv from "dotenv";
+import jwt from "jsonwebtoken";
+import {throwInternalError} from "../middleware/error.middleware";
+import {isAddress, verifyMessage} from "viem";
+import {dataSource} from "../db";
+import {AddressesEntity} from "../entities/addreesses.entity";
+import {UsersEntity} from "../entities/users.entity";
+import {v4 as uuidv4} from 'uuid'
+
+dotenv.config();
+
+const router = express.Router();
+
+
+const addressesRepo = dataSource.getRepository(AddressesEntity)
+const usersRepo = dataSource.getRepository(UsersEntity)
+
+router.post('/login', async (req: Request, res: Response) => {
+  const {signature, walletAddress, challengeMessage} = req.body;
+
+  if (!isAddress(walletAddress)) {
+    res.status(403).json({error: "Invalid wallet address"})
+    return
+  }
+
+  try {
+    const address = await verifyMessage({
```

Comment:
We could use:  [verifySiweMessage](https://viem.sh/docs/siwe/actions/verifySiweMessage). We would need a public client somewhere in a config file though. Like:
```ts
// config/client.ts
import { createPublicClient, http } from 'viem'
import { mainnet } from 'viem/chains'
 
export const publicClient = createPublicClient({
  chain: mainnet,
  transport: http() // Add an RPC URL
})
```
This is a more standard way of logging in to a web3 app like ours. Since the user has already connected their wallet to the extension a walletClient is available to generate a SIWE message on the front and send both signature and challengeMessage to this endpoint. Need feedback from @SolutionsEngineer, @lennardevertz, or @geoist1 

Also the name `address` does not match the boolean return type. I would use `valid` instead.

Reply:
yeah, I think @lennardevertz, suggested that will be better to use SIWE standard

Reply:
I added verifySiweMessage

Reply:
Reverted. This will be added in a separate PR from me.